maintenance/health/servicehealthcheck/README.md

```diff
@@ -1,7 +1,24 @@
-# Health Checks for Services 
-* Makes it possible to have a health check for different services (e.g. postgres, redis)
+# Health Check
+* Makes it possible to add a health check for a service (e.g. postgres) to the list of checks checked when for `/health` and `/health/check`
 
-* A service has to implement the `HealthCheck` interface and register the health check with `servicehealthchecker.RegisterHealthCheck(hc HealthCheck, name string)`
+* `/health` returns OK(200) if all registered and required health checks are healthy and ERR(503) if any health check is not healthy
 
-* `healthchecker` does NOT cache the results but offers a struct (`ConnectionState`) for caching 
+* `/health/check` returns a table with the results of all registered health checks
 
+* Each check has to implement the `HealthCheck` interface and has to be registered
+    * health checks are registered as required or optional 
+    * `servicehealthcheck.RegisterHealthCheck(hc HealthCheck, name string)` registers a health check with a unique name as required check
+    * `servicehealthcheck.RegisterOptionalHealthCheck(hc HealthCheck, name string)` registers a health check with a unique name as optional check
+    * a optional check is not checked when `/health` is called
+    * a required check is always checked
+    
+* To make a optional health check required (or a required health check optional) the check has to be removed and registered again. This should't happen very often.
```

Comment:
1. `an` instead of `a`. 
2. What does the last sentence actually mean, or what is the motivation for putting it there? That the user should not switch checks too often between required and optional? Is that even a thing that someone would da?